src/utils/Storage.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Helper functions to store and retrieve cache data.
+ */
+
+/**
+ * Store a value.
+ *
+ * @param key - A unique string.
+ * @param value - The value to store. Can be anything that is JSON serializable.
+ * @param timeout - The number of seconds from now for this value to be valid.
+ *                  If undefined, this value will never expire.
+ * @param user - If set, associate this value with a user.
+ */
+export function store(
+  key: string,
+  value: any,
+  timeout?: number,
+  user?: string
+): any {
+  try {
+    const data = {
+      value,
+      user,
+      ...(timeout && { expiration: `${Date.now() + timeout * 1000}` }),
+    }
+
+    window.localStorage.setItem(key, JSON.stringify(data))
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err)
```

Comment:
Can we avoid using console if possible, even if it is error. Generally we just want to avoid outputting any console oriented messages to the user directly.

Reply:
That makes sense! Will remove them later today.

Reply:
Should we just ignore the errors then?

Reply:
I feel like we should address errors in some way. Do you think it would be better to return it as a promise and reject if it fails in any way? We could also just return null, default, or some error response in this situation for the function calling this.

Reply:
It makes sense to not log things to console in a production environment. It _might_ be helpful during development for some situations (but probably not in this case).

I think it's fine to ignore them here - we don't really need to return anything at all for the `store` function, and we'll be returning the default for the function below. If these fail for some reason (e.g.: localStorage is not supported), we just won't use localStorage.